airbyte-config/specs/src/main/java/io/airbyte/config/specs/SeedConnectorSpecGenerator.java

```diff
@@ -0,0 +1,133 @@
+ */
+public class SeedConnectorSpecGenerator {
+
+  private static final String DOCKER_REPOSITORY_FIELD = "dockerRepository";
+  private static final String DOCKER_IMAGE_TAG_FIELD = "dockerImageTag";
+  private static final String DOCKER_IMAGE_FIELD = "dockerImage";
+  private static final String SPEC_FIELD = "spec";
+  private static final String SPEC_BUCKET_NAME = new EnvConfigs().getSpecCacheBucket();
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SeedConnectorSpecGenerator.class);
+
+  private static final Options OPTIONS = new Options();
+  private static final Option SEED_ROOT_OPTION = new Option("s", "seed-root", true, "path to where seed resource files are stored");
+
+  private final GcsBucketSpecFetcher bucketSpecFetcher;
+
+  SeedConnectorSpecGenerator(final GcsBucketSpecFetcher bucketSpecFetcher) {
+    this.bucketSpecFetcher = bucketSpecFetcher;
+  }
+
+  static {
+    SEED_ROOT_OPTION.setRequired(true);
+    OPTIONS.addOption(SEED_ROOT_OPTION);
+  }
+
+  public static void main(final String[] args) throws Exception {
+    final CommandLine parsed = parse(args);
+    final Path outputRoot = Path.of(parsed.getOptionValue(SEED_ROOT_OPTION.getOpt()));
+
+    final GcsBucketSpecFetcher bucketSpecFetcher = new GcsBucketSpecFetcher(StorageOptions.getDefaultInstance().getService(), SPEC_BUCKET_NAME);
+    final SeedConnectorSpecGenerator seedConnectorSpecGenerator = new SeedConnectorSpecGenerator(bucketSpecFetcher);
+    seedConnectorSpecGenerator.run(outputRoot, ConnectorType.SOURCE);
+    seedConnectorSpecGenerator.run(outputRoot, ConnectorType.DESTINATION);
+  }
+
+  public void run(final Path seedRoot, final ConnectorType connectorType) throws IOException {
+    LOGGER.info("Updating seeded {} definition specs if necessary...", connectorType.name());
+
+    final JsonNode seedDefinitionsJson = yamlToJson(seedRoot, connectorType.getDefinitionFileName());
+    final JsonNode seedSpecsJson = yamlToJson(seedRoot, connectorType.getSpecFileName());
+
+    final List<DockerImageSpec> updatedSeedSpecs = fetchUpdatedSeedSpecs(seedDefinitionsJson, seedSpecsJson);
+
+    final String outputString = String.format("# This file is generated by %s.\n", this.getClass().getName())
+        + "# Do NOT edit this file directly. See generator class for more details.\n"
+        + Yamls.serialize(updatedSeedSpecs);
+    final Path outputPath = IOs.writeFile(seedRoot.resolve(connectorType.getSpecFileName()), outputString);
+
+    LOGGER.info("Finished updating {}", outputPath);
+  }
+
+  private JsonNode yamlToJson(final Path root, final String fileName) {
+    final String yamlString = IOs.readFile(root, fileName);
+    return Yamls.deserialize(yamlString);
+  }
+
+  @VisibleForTesting
+  final List<DockerImageSpec> fetchUpdatedSeedSpecs(final JsonNode seedDefinitions, final JsonNode currentSeedSpecs) {
+    final List<String> seedDefinitionsDockerImages = MoreIterators.toList(seedDefinitions.elements()).stream()
```

Comment:
by convention we generally put the `.stream` on the new line by itself when a stream chain is going to be broken up over multiple lines.